### DISCUSSION TITLE
Building jsqlparser 4.7 from sources

### DISCUSSION THREAD

**Discussion by mikatane** [JSQLParser/JSqlParser#1881] [Q&A]
I want to build from the java source code (with some extensions to the grammar) -- I hope to build by setting up an eclipse project, though that is not essential to me.

The source code contains references to types Provider, StreamProvider, StringProvider that are not included in the zip file I get from github, nor are they generated by jjtree/javacc

**Comment by manticore-projects** [comment]
Greetings,

JSQLParser sits on JavaCC (a parser generator) which builds the Parser from a Grammar. Thus there are two steps needed:
1) generate the parser sources from the Grammar
2) build the parser

The Maven/Gradle plugins handle this gracefully. After generating the parser sources, you can find them in `JSQLParser/build/generated/javacc/net/sf/jsqlparser/parser`:

```bash
are@ryzen ~/d/s/J/JSQLParser (master)> ls -lh build/generated/javacc/net/sf/jsqlparser/parser
total 1.3M
-rw-r--r-- 1 are users  29K Oct 21 10:05 CCJSqlParserConstants.java
-rw-r--r-- 1 are users 976K Oct 21 10:05 CCJSqlParser.java
-rw-r--r-- 1 are users 284K Oct 21 10:05 CCJSqlParserTokenManager.java
-rw-r--r-- 1 are users 7.0K Oct 21 10:05 ParseException.java
-rw-r--r-- 1 are users  893 Oct 21 10:05 Provider.java
-rw-r--r-- 1 are users 1.8K Oct 21 10:05 StreamProvider.java
-rw-r--r-- 1 are users 1.2K Oct 21 10:05 StringProvider.java
-rw-r--r-- 1 are users 4.1K Oct 21 10:05 Token.java
-rw-r--r-- 1 are users 4.5K Oct 21 10:05 TokenMgrError.java
```
